### ISSUE TITLE
Build failed with MSVC sln generated from CMAKE

### ISSUE THREAD

**Issue by Asperger** [alanxz/rabbitmq-c#532]
Trying to build shared library from master branch with VS2015. The system is Windows 10 version 1803 with OS build 17134.285. The CMAKE config is [here](https://i.imgur.com/bY4hczN.png). OpenSSL comes from [here](https://code.google.com/archive/p/openssl-for-windows/downloads).
Then I get the following error message:
>2>amqp_openssl.obj : error LNK2019: unresolved external symbol SSL_CTX_clear_options referenced in function amqp_ssl_socket_set_ssl_versions
>2>amqp_openssl.obj : error LNK2019: unresolved external symbol FIPS_mode_set referenced in function amqp_uninitialize_ssl_library
>2>D:\GitHub\rabbitmq-c\build-x64\librabbitmq\Release\rabbitmq.4.dll : fatal error LNK1120: 2 unresolved externals

while the static library looks good so far.
Is there any way to fix that or I have to stick with the static library?

**Comment by sigiesec** [comment]
I think the versions from openssl-for-windows are very outdated. While the README of rabbitmq-c claims that 0.9.8 is supported, I think that at least 0.9.8m is required. However, you should not use any 0.9.8 version, since it has been out of support for quite some time:
```
Note: The latest stable version is the 1.1.1 series. This is also our Long Term Support (LTS) version, supported until 11th September 2023. Our previous LTS version (1.0.2 series) will continue to be supported until 31st December 2019 (security fixes only during the last year of support). The 1.1.0 series is currently only receiving security fixes and will go out of support on 11th September 2019. All users of 1.0.2 and 1.1.0 are encouraged to upgrade to 1.1.1 as soon as possible. The 0.9.8, 1.0.0 and 1.0.1 versions are now out of support and should not be used.
```
(from https://www.openssl.org/source/)

Building a static library does not include a link step, so this is probably the reason you don't get a linker error. If you have the matching headers, there should be a compiler error with the old version, but it is probably not worthwhile to investigate the details of this.

**Comment by Asperger** [comment]
Problem solved with openssl 1.1.1. Thanks!
